pkg/controller/bundle/bundle_unpacker.go

```diff
@@ -165,26 +185,62 @@ func (c *ConfigMapUnpacker) job(cmRef *corev1.ObjectReference, bundlePath string
 	job.SetName(cmRef.Name)
 	job.SetOwnerReferences([]metav1.OwnerReference{ownerRef(cmRef)})
 
+	// By default the BackoffLimit is set to 6 which with exponential backoff 10s + 20s + 40s ...
+	// translates to ~10m of waiting time.
+	// We want to fail faster than that when we have repeated failures from the bundle unpack pod
+	// so we set it to 3 which is ~1m of waiting time
+	// See: https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-backoff-failure-policy
+	// TODO (haseeb): Should this be configurable as well?
+	backOffLimit := int32(3)
+	job.Spec.BackoffLimit = &backOffLimit
+
+	// Set ActiveDeadlineSeconds as the unpack timeout
+	// Don't set a timeout if it is 0
+	if c.unpackTimeout != time.Duration(0) {
+		t := int64(c.unpackTimeout.Seconds())
+		job.Spec.ActiveDeadlineSeconds = &t
+	}
+
+	// The bundle timeout annotation if specified overrides the --bundle-unpack-timeout flag value
+	timeoutStr, ok := ipAnnotations[BundleUnpackTimeoutAnnotationKey]
+	if !ok {
+		return job
+	}
+	d, err := time.ParseDuration(timeoutStr)
+	if err != nil {
+		// TODO(haseeb): log this error
+		return job
+	}
+	if d == time.Duration(0) {
+		// 0 means no timeout
+		return job
+	}
+
+	timeoutSeconds := int64(d.Seconds())
+	job.Spec.ActiveDeadlineSeconds = &timeoutSeconds
+
 	return job
 }
 
 //go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . Unpacker
 
 type Unpacker interface {
-	UnpackBundle(lookup *operatorsv1alpha1.BundleLookup) (result *BundleUnpackResult, err error)
+	UnpackBundle(lookup *operatorsv1alpha1.BundleLookup, ipAnnotations map[string]string) (result *BundleUnpackResult, err error)
```

Comment:
I think this should have a `timeout` parameter instead. It's unnecessary to depend on the entire annotations map.

Reply:
Updated: We're parsing the timeout annotation before calling UnpackBundle() now so we don't pass in the entire annotation set anymore.